### ISSUE TITLE
ValueError when running rl_with_open_ai_gym_wrapper

### ISSUE THREAD

**Issue by cjyu81** [hsahovic/poke-env#58]
I am up to date with the master branch of Keras and Keras-RL. I used the two commands
pip install git+git://github.com/wau/keras-rl2.git --upgrade --no-deps
pip install git+git://github.com/fchollet/keras.git --upgrade --no-deps

However, when running python rl_with_open_ai_gym_wrapper.py, I get:

Exception in thread Thread-1:
Traceback (most recent call last):
  File "C:\Users\14086\anaconda3\envs\tensorflow\lib\threading.py", line 926, in _bootstrap_inner
    self.run()
  File "C:\Users\14086\anaconda3\envs\tensorflow\lib\threading.py", line 870, in run
    self._target(*self._args, **self._kwargs)
  File "C:\Users\14086\anaconda3\envs\tensorflow\lib\site-packages\poke_env\player\env_player.py", line 344, in <lambda>
    target=lambda: env_algorithm_wrapper(self, env_algorithm_kwargs)
  File "C:\Users\14086\anaconda3\envs\tensorflow\lib\site-packages\poke_env\player\env_player.py", line 328, in env_algorithm_wrapper
    env_algorithm(player, **kwargs)
  File "rl_with_open_ai_gym_wrapper.py", line 79, in dqn_training
    dqn.fit(player, nb_steps=nb_steps)
  File "C:\Users\14086\anaconda3\envs\tensorflow\lib\site-packages\rl\core.py", line 169, in fit
    action = self.forward(observation)
  File "C:\Users\14086\anaconda3\envs\tensorflow\lib\site-packages\rl\agents\dqn.py", line 225, in forward
    q_values = self.compute_q_values(state)
  File "C:\Users\14086\anaconda3\envs\tensorflow\lib\site-packages\rl\agents\dqn.py", line 69, in compute_q_values
    q_values = self.compute_batch_q_values([state]).flatten()
  File "C:\Users\14086\anaconda3\envs\tensorflow\lib\site-packages\rl\agents\dqn.py", line 64, in compute_batch_q_values
    q_values = self.model.predict_on_batch(batch)
  File "C:\Users\14086\anaconda3\envs\tensorflow\lib\site-packages\tensorflow_core\python\keras\engine\training.py", line 1261, in predict_on_batch
    self._make_predict_function()
  File "C:\Users\14086\anaconda3\envs\tensorflow\lib\site-packages\tensorflow_core\python\keras\engine\training.py", line 2121, in _make_predict_function
    **kwargs)
  File "C:\Users\14086\anaconda3\envs\tensorflow\lib\site-packages\tensorflow_core\python\keras\backend.py", line 3769, in function
    return GraphExecutionFunction(inputs, outputs, updates=updates, **kwargs)
  File "C:\Users\14086\anaconda3\envs\tensorflow\lib\site-packages\tensorflow_core\python\keras\backend.py", line 3421, in __init__
    with ops.control_dependencies([self.outputs[0]]):
  File "C:\Users\14086\anaconda3\envs\tensorflow\lib\site-packages\tensorflow_core\python\framework\ops.py", line 5159, in control_dependencies
    return get_default_graph().control_dependencies(control_inputs)
  File "C:\Users\14086\anaconda3\envs\tensorflow\lib\site-packages\tensorflow_core\python\framework\ops.py", line 4583, in control_dependencies
    c = self.as_graph_element(c)
  File "C:\Users\14086\anaconda3\envs\tensorflow\lib\site-packages\tensorflow_core\python\framework\ops.py", line 3505, in as_graph_element
    return self._as_graph_element_locked(obj, allow_tensor, allow_operation)
  File "C:\Users\14086\anaconda3\envs\tensorflow\lib\site-packages\tensorflow_core\python\framework\ops.py", line 3584, in _as_graph_element_locked
    raise ValueError("Tensor %s is not an element of this graph." % obj)
ValueError: Tensor Tensor("dense_2/BiasAdd:0", shape=(None, 22), dtype=float32) is not an element of this graph.

What should I do about this?

**Comment by cjyu81** [comment]
Whenever I run the command, a new battle opens in my local server, but the bots don't make any moves; it's eternally "Waiting for players..."

![image](https://user-images.githubusercontent.com/39837939/89236267-e87bc600-d5a4-11ea-8142-0b1662f1766a.png)

**Comment by hsahovic** [comment]
Hey @cjyu81 ! Thanks for opening this issue.

[keras-rl2](https://github.com/wau/keras-rl2) requires `tensorflow==2.1.0`.

Can you run:

```
pip uninstall tensorflow
pip install tensorflow==2.1.0
```

And see if that helps?

**Comment by cjyu81** [comment]
I uninstalled tensorflow(it was 2.1.0) and reinstalled as you said but it didn't fix the problem.

**Comment by hsahovic** [comment]
After doing some experimenting in a fresh environment, I realized that this is actually a problem we encountered before: it looks like the latest version of `keras-rl2`, version `1.0.4`, is not fully backward compatible with version `1.0.3`.

Reverting to version `1.0.3` should solve the problem. Running the following:
```
pip uninstall --yes tensorflow
pip uninstall --yes keras
pip uninstall --yes keras-rl
pip uninstall --yes keras-rl2
pip install tensorflow==2.1.0
pip install keras-rl2==1.0.3
```
Should do the trick.

Let me know if you are able to run the example with these versions!
